core/src/main/java/org/bitcoinj/crypto/HDPath.java

```diff
@@ -172,13 +174,13 @@ public static HDPath m(ChildNumber... children) {
      */
     public static HDPath parsePath(@Nonnull String path) {
         List<String> parsedNodes = SEPARATOR_SPLITTER.splitToList(path);
-        boolean hasPrivateKey = false;
-        if (!parsedNodes.isEmpty()) {
-            final String firstNode = parsedNodes.get(0);
-            if (firstNode.equals(Character.toString(PREFIX_PRIVATE)))
-                hasPrivateKey = true;
-            if (hasPrivateKey || firstNode.equals(Character.toString(PREFIX_PUBLIC)))
-                parsedNodes.remove(0);
+
+        String firstNode = parsedNodes.get(0);
```

Comment:
How do you know the list isn't empty at this point?

Reply:
We don't. But the prefix comparisons that follow work correctly if `firstNode` is `null`.

Reply:
If parserNodes is an empty list, it will throw at .get(0)?

Reply:
Oops. I wasn't thinking correctly, I was thinking it would return `null` for an empty `List`.

One could argue that an empty string (and thus an empty `parsedNode` list) is invalid and _should_ throw an exception, but that would be a breaking change and would break current unit tests.

Let me rethink this.

Reply:
Update: it didn't require much rethinking, just a test for empty. Fixed and force-pushed.